testing/unit/munit-mockerrinj-c.cpp

```diff
@@ -0,0 +1,277 @@
+	printf("MOCK " fmt "\n", ## __VA_ARGS__)
+
+using namespace opae::testing;
+
+
+static int port_release_ioctl(mock_object * m, int request, va_list argp){
+  int retval = -1;
+  errno = EINVAL;
+  struct fpga_fme_port_release *port_release =
+         va_arg(argp, struct fpga_fme_port_release *);
+  if (!port_release) {
+    FPGA_MSG("port_release is NULL");
+    goto out_EINVAL;
+  }
+  if (port_release->argsz != sizeof(*port_release)) {
+    FPGA_MSG("wrong structure size");
+    goto out_EINVAL;
+  }
+  if (port_release->flags != 0) {
+    FPGA_MSG("unexpected flags %u", port_release->flags);
+    goto out_EINVAL;
+  }
+  if (port_release->port_id != 0) {
+    FPGA_MSG("unexpected port ID %u", port_release->port_id);
+    goto out_EINVAL;
+  }
+  retval = 0;
+  errno = 0;
+
+out:
+  va_end(argp);
+  return retval;
+
+out_EINVAL:
+  retval = -1;
+  errno = EINVAL;
+  goto out;
+}
+
+static int port_assign_ioctl(mock_object * m, int request, va_list argp){
+  int retval = -1;
+  errno = EINVAL;
+  struct fpga_fme_port_assign *port_assign =
+      va_arg(argp, struct fpga_fme_port_assign *);
+  if (!port_assign) {
+      FPGA_MSG("port_assign is NULL");
+      goto out_EINVAL;
+  }
+  if (port_assign->argsz != sizeof(*port_assign)) {
+      FPGA_MSG("wrong structure size");
+      goto out_EINVAL;
+  }
+  if (port_assign->flags != 0) {
+      FPGA_MSG("unexpected flags %u", port_assign->flags);
+      goto out_EINVAL;
+  }
+  if (port_assign->port_id != 0) {
+      FPGA_MSG("unexpected port ID %u", port_assign->port_id);
+      goto out_EINVAL;
+  }
+retval = 0;
+errno = 0;
+out:
+  va_end(argp);
+  return retval;
+
+out_EINVAL:
+  retval = -1;
+  errno = EINVAL;
+  goto out;
+}
+
+
+
+class mock_err_inj_c_p : public ::testing::TestWithParam<std::string> {
+ protected:
+   mock_err_inj_c_p() : tmpsysfs_("mocksys-XXXXXX") {}
+
+  virtual void SetUp() override {
+    ASSERT_TRUE(test_platform::exists(GetParam()));
+    platform_ = test_platform::get(GetParam());
+    system_ = test_system::instance();
+    system_->initialize();
+    tmpsysfs_ = system_->prepare_syfs(platform_);
+
+    ASSERT_EQ(fpgaGetProperties(nullptr, &filter_), FPGA_OK);
+    ASSERT_EQ(fpgaPropertiesSetObjectType(filter_, FPGA_DEVICE), FPGA_OK);
+    ASSERT_EQ(fpgaEnumerate(&filter_, 1, tokens_.data(), tokens_.size(),
+                            &num_matches_),
+              FPGA_OK);
+  }
+
+  virtual void TearDown() override {
+    EXPECT_EQ(fpgaDestroyProperties(&filter_), FPGA_OK);
+    if (!tmpsysfs_.empty() && tmpsysfs_.size() > 1) {
+      std::string cmd = "rm -rf " + tmpsysfs_;
+      std::system(cmd.c_str());
+    }
+    system_->finalize();
+  }
+
+  std::string tmpsysfs_;
+  fpga_properties filter_;
+  std::array<fpga_token, 2> tokens_;
+  fpga_handle handle_;
+  uint32_t num_matches_;
+  test_platform platform_;
+  test_system *system_;
+};
+
+
+/**
+* @test    fpga_mock_errinj_01
+* @brief   Tests: fpgaReset
+* @details fpgaReset resets fpga afu
```

Comment:
It's not clear from the wording what this test is trying to accomplish.  What is the error being injected?